File: crates/varde-code/tests/fixtures/php/control_error.php
<?php
// Control-flow / error fixtures: ControlFlow (if/foreach/while/switch/try/
// return/break), Catch (catch_clause), and Throw (throw_expression), plus the
// calls and member accesses that keep the fixture self-describing.

function process($items)
{
    $total = 0;
    if ($items) {
        try {
            risky();
        } catch (\Exception $e) {
            throw new RuntimeException("boom");
        }
        foreach ($items as $n) {
            $total = $total + $n;
        }
        while ($total > 0) {
            $total = $total - 1;
        }
        switch ($total) {
            case 0:
                break;
            default:
                break;
        }
    }
    return $total;
}

File: crates/varde-code/tests/fixtures/scala/control_error.scala
// Control-flow / error fixtures: ControlFlow (if/while/for/match/try/return),
// Catch (`catch { case e => }`), and Throw (`throw new ...`), plus calls and
// literals to keep the fixture self-describing.

object Processor {
  def process(items: List[Int]): String = {
    var total = 0
    if (items.isEmpty) {
      return "none"
    }
    for (item <- items) {
      while (total < item) {
        total = total + 1
      }
    }
    val label = total match {
      case 0 => "zero"
      case _ => "some"
    }
    try {
      throw new IllegalStateException("boom")
    } catch {
      case e: Exception => println(e)
    }
    label
  }
}
